fv3core/pace/fv3core/initialization/geos_wrapper.py

```diff
@@ -0,0 +1,420 @@
+        )
+        quantity_factory = pace.util.QuantityFactory.from_backend(
+            sizer=sizer, backend=backend
+        )
+
+        # set up the metric terms and grid data
+        metric_terms = pace.util.grid.MetricTerms(
+            quantity_factory=quantity_factory, communicator=self.communicator
+        )
+        grid_data = pace.util.grid.GridData.new_from_metric_terms(metric_terms)
+
+        stencil_config = pace.dsl.stencil.StencilConfig(
+            compilation_config=pace.dsl.stencil.CompilationConfig(
+                backend=backend, rebuild=False, validate_args=False
+            ),
+        )
+
+        self._grid_indexing = pace.dsl.stencil.GridIndexing.from_sizer_and_communicator(
+            sizer=sizer, cube=self.communicator
+        )
+        stencil_factory = pace.dsl.StencilFactory(
+            config=stencil_config, grid_indexing=self._grid_indexing
+        )
+
+        self.dycore_state = fv3core.DycoreState.init_zeros(
+            quantity_factory=quantity_factory
+        )
+
+        self.dycore_state.bdt = float(namelist["dt_atmos"])
+        if "fv_core_nml" in namelist.keys():
+            self.dycore_state.bdt = (
+                float(namelist["dt_atmos"]) / namelist["fv_core_nml"]["k_split"]
+            )
+        elif "dycore_config" in namelist.keys():
+            self.dycore_state.bdt = (
+                float(namelist["dt_atmos"]) / namelist["dycore_config"]["k_split"]
+            )
+        else:
+            raise KeyError("Cannot find k_split in namelist")
+
+        damping_coefficients = pace.util.grid.DampingCoefficients.new_from_metric_terms(
+            metric_terms
+        )
+
+        self.dynamical_core = fv3core.DynamicalCore(
+            comm=self.communicator,
+            grid_data=grid_data,
+            stencil_factory=stencil_factory,
+            quantity_factory=quantity_factory,
+            damping_coefficients=damping_coefficients,
+            config=self.dycore_config,
+            timestep=timedelta(seconds=self.dycore_config.dt_atmos),
+            phis=self.dycore_state.phis,
+            state=self.dycore_state,
+        )
+
+        self.output_dictionary: Dict[str, np.ndarray] = {}
+        self._allocate_output_dir()
+
+    def __call__(
+        self,
+        u: np.ndarray,
+        v: np.ndarray,
+        w: np.ndarray,
+        delz: np.ndarray,
+        pt: np.ndarray,
+        delp: np.ndarray,
+        q: np.ndarray,
+        ps: np.ndarray,
+        pe: np.ndarray,
+        pk: np.ndarray,
+        peln: np.ndarray,
+        pkz: np.ndarray,
+        phis: np.ndarray,
+        q_con: np.ndarray,
+        omga: np.ndarray,
+        ua: np.ndarray,
+        va: np.ndarray,
+        uc: np.ndarray,
+        vc: np.ndarray,
+        mfxd: np.ndarray,
+        mfyd: np.ndarray,
+        cxd: np.ndarray,
+        cyd: np.ndarray,
+        diss_estd: np.ndarray,
+    ) -> dict:
+
+        self._put_fortran_data_in_dycore(
```

Comment:
Grab the timer from `pace.util` and slap one around each of those function call. See `fv_dynamics` for usage. They are GPU robust already